source/module_hamilt_general/module_xc/xc_functional_libxc_wrapper_xc.cpp

```diff
@@ -0,0 +1,38 @@
+#ifdef USE_LIBXC
+
+#include "xc_functional_libxc.h"
+
+void XC_Functional_Libxc::xc_spin_libxc(
+        const std::vector<int> &func_id,
+        const double &rhoup, const double &rhodw,
+		double &exc, double &vxcup, double &vxcdw)
+{
+    double e, vup, vdw;
+    double *rho_ud, *vxc_ud;
+    exc = vxcup = vxcdw = 0.0;
+
+    rho_ud = new double[2];
```

Comment:
will it be possible to use std::vector instead of new and delete?

Reply:
updated